pkg/reconciler/taskrun/taskrun.go

```diff
@@ -970,3 +970,104 @@ type Host struct {
 func platformLabel(platform string) string {
 	return strings.ReplaceAll(platform, "/", "-")
 }
+
+// UpdateTaskRunWithRetry performs a conflict-resilient update of a TaskRun object.
+// On conflict, it fetches the latest version and merges labels, annotations, and finalizers from the incoming TaskRun.
+func UpdateTaskRunWithRetry(ctx context.Context, client client.Client, apiReader client.Reader, tr *tektonapi.TaskRun, maxRetries int) error {
```

Comment:
We could use the [retry logic](https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict) available from the client-go library. It seems to me that we could simplify our code and offload the implementation of the retry logic to the kubernetes toolchain. Do you think it is a good fit here?

Reply:
assuming we move the update logic to a function called `updateTaskRun` it become something like follows:

```golang
func UpdateTaskRunWithRetry(ctx context.Context, client client.Client, apiReader client.Reader, tr *tektonapi.TaskRun, maxRetries int) error {
	  return retry.RetryOnConflict(retry.DefaultRetry, func() error { 
	  	  return updateTaskRun(ctx, client, apiReader, tr)
	  }
}
```